bindings/python/pymongoarrow/context.py

```diff
@@ -41,21 +43,27 @@ def __init__(self, schema, builder_map, type_map):
         self.type_map = type_map
 
     @classmethod
-    def from_schema(cls, schema):
+    def from_schema(cls, schema, codec_options=DEFAULT_CODEC_OPTIONS):
         """Initialize the context from a :class:`~pymongoarrow.schema.Schema`
         instance.
 
         :Parameters:
           - `schema`: Instance of :class:`~pymongoarrow.schema.Schema`.
+          - `codec_options` (optional): An instance of
+            :class:`~bson.codec_options.CodecOptions`.
         """
         builder_map = {}
         type_map = {}
         str_type_map = _get_internal_typemap(schema.typemap)
         for fname, ftype in str_type_map.items():
             builder_cls = _TYPE_TO_BUILDER_CLS[ftype]
             encoded_fname = fname.encode('utf-8')
+            # special-case initializing builders for parameterized types
             if builder_cls == DatetimeBuilder:
                 arrow_type = schema.typemap[fname]
+                if codec_options.tzinfo is not None and arrow_type.tz is None:
```

Comment:
Should we need to check codec_options.tz_aware here too? That would make it match pymongo's decoding behavior more closely.

Reply:
We can, but its not possible to create CodecOptions instances with non-`None` `tzinfo` if `tz_aware` is not also set to `True` so I am not sure it adds much value:
```
CodecOptions(tzinfo=datetime.timezone.utc)
Traceback (most recent call last):
  File "/usr/local/Caskroom/miniconda/base/envs/arrow/lib/python3.7/site-packages/IPython/core/interactiveshell.py", line 3437, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-60-6927577ac152>", line 1, in <module>
    CodecOptions(tzinfo=datetime.timezone.utc)
  File "/usr/local/Caskroom/miniconda/base/envs/arrow/lib/python3.7/site-packages/bson/codec_options.py", line 271, in __new__
    "cannot specify tzinfo without also setting tz_aware=True")
ValueError: cannot specify tzinfo without also setting tz_aware=True
```

Reply:
Ah thanks for checking that.